### PR TITLE
Graff46 season

### PR DESCRIPTION
**Pull request by Graff46**
Тест смены времён года.
Нужно проверить код.

### CODE REVIEW

File: code/engine.vc2008/xrEngine/IGame_Level.cpp [hunk 1]
```diff
@@ -11,18 +11,20 @@
 #include "CameraManager.h"
 #include "xr_object.h"
 #include "feel_sound.h"
+#include "../xrCore/season.h"
 
-ENGINE_API	IGame_Level*	g_pGameLevel	= nullptr;
+ENGINE_API	IGame_Level*	g_pGameLevel	= NULL;
```
Comment (Xottab-DUTY) [inline code/engine.vc2008/xrEngine/IGame_Level.cpp]: Why?

File: code/engine.vc2008/xrEngine/IGame_Level.cpp [hunk 2]
```diff
@@ -131,6 +141,7 @@ BOOL IGame_Level::Load			(u32 dwNum)
 #include "../xrCore/threadpool/ttapi.h"
 #endif
 
+int		psNET_DedicatedSleep	= 5;
```
Comment (Xottab-DUTY) [inline code/engine.vc2008/xrEngine/IGame_Level.cpp]: If I recall correctly, this was deleted..

File: code/engine.vc2008/xrEngine/x_ray.cpp [hunk 3]
```diff
@@ -374,16 +375,15 @@ struct damn_keys_filter {
 
 #include "xr_ioc_cmd.h"
 
-ENGINE_API int RunApplication(LPCSTR commandLine)
+ENGINE_API int RunApplication(char* commandLine)
```
Comment (Xottab-DUTY) [inline code/engine.vc2008/xrEngine/x_ray.cpp]: Some people from Oxygen Team won't like it.

File: code/engine.vc2008/xrEngine/x_ray.cpp [hunk 4]
```diff
@@ -374,16 +375,15 @@ struct damn_keys_filter {
...
 {
 	if (!IsDebuggerPresent()) 
 	{
 		size_t HeapFragValue = 2;
 		HeapSetInformation(GetProcessHeap(), HeapCompatibilityInformation, &HeapFragValue, sizeof(HeapFragValue));
 	}
 
-    gMainThreadId = GetCurrentThreadId();
-    Debug.set_mainThreadId(gMainThreadId);
```
Comment (Xottab-DUTY) [inline code/engine.vc2008/xrEngine/x_ray.cpp]: Don't delete this.

File: code/engine.vc2008/xrGame/Level_network_map_sync.cpp [hunk 5]
```diff
@@ -2,39 +2,66 @@
...
-
+	
 	if (!map_data.m_map_sync_received)
 	{
 		Sleep(5);
 		++map_data.m_wait_map_time;
 		return false;
 	}
-
+
```
Comment (Xottab-DUTY) [inline code/engine.vc2008/xrGame/Level_network_map_sync.cpp]: Useless tabs...

File: code/engine.vc2008/xrGame/Level_network_map_sync.cpp [hunk 6]
```diff
@@ -72,11 +99,8 @@ bool	CLevel::synchronize_client()
 	{
 		ClientReceive();
 		Server->Update();
-
-        //Don't delete this ClientReceive(). Without that, game_configured variable will not changed
-        ClientReceive();
```
Comment (Xottab-DUTY) [inline code/engine.vc2008/xrGame/Level_network_map_sync.cpp]:
```cpp
//Don't delete this ClientReceive(). Without that, game_configured variable will not changed
```

File: code/xrEngine.sln [hunk 7]
```diff
@@ -1,286 +1,286 @@
-Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (Xottab-DUTY) [inline code/xrEngine.sln]: =(

**Comment by ForserX** [comment]
Невалидный PR. Ломает всё.

**Pull request closed without merging**